src/matrix/Client.ts

```diff
@@ -485,33 +514,51 @@ export class Client {
             // if one fails, don't block the other from trying
             // also, run in parallel
             await Promise.all([
-                log.wrap("storageFactory", () => this._platform.storageFactory.delete(this._sessionId)),
-                log.wrap("sessionInfoStorage", () => this._platform.sessionInfoStorage.delete(this._sessionId)),
+                log?.wrap("storageFactory", () => this._platform.storageFactory.delete(this._sessionId)),
+                log?.wrap("sessionInfoStorage", () => this._platform.sessionInfoStorage.delete(this._sessionId)),
             ]);
-            this._sessionId = null;
+            this._sessionId = undefined;
         }
     }
 
     _resetStatus() {
         this._status.set(LoadStatus.NotLoading);
-        this._error = null;
-        this._loginFailure = null;
+        this._error = undefined;
+        this._loginFailure = undefined;
     }
 }
 
 class AccountSetup {
-    constructor(encryptedDehydratedDevice, finishStage) {
+    _encryptedDehydratedDevice: EncryptedDehydratedDevice;
+    _dehydratedDevice?: any;
```

Comment:
This appears to only get set by `finish(dehydratedDevice: any)` down below, but afaict `finish` never gets called. Maybe worth a cleanup here, but I don't understand the intention well enough to do so myself.